theseus/optimizer/autograd/sparse_autograd.py

```diff
@@ -12,8 +12,9 @@
 class CholmodSolveFunction(torch.autograd.Function):
     @staticmethod
     def forward(ctx, *args, **kwargs):
-        At_val: torch.Tensor = args[0]
-        b: torch.Tensor = args[1]
+        dtype_original = args[0].dtype
```

Comment:
Maybe leave a comment about what's happening here and why (along the lines of: we are casting to double and back for numerical stability).